backend/tests/currentUser.spec.js

```diff
@@ -0,0 +1,75 @@
+require('dotenv').config({ path: './.env' });
+const jwt = require('jsonwebtoken');
+const { CrudServer } = require('../src/server');
+const request = require('supertest');
+const { assert, expect } = require('chai');
+const User = require('../src/api/users/user.model');
+
+describe('CurrentUser test suite', () => {
+  let server;
+
+  before(async () => {
+    const crudServer = new CrudServer();
+    await crudServer.start();
+    server = crudServer.server;
+  });
+
+  describe('GET /users/current', () => {
+    context('when bad token was provided', () => {
+      let response;
+
+      before(async () => {
+        response = await request(server)
+          .get('/api/v1/users/current')
+          .set('Authorization', 'Bearer bad_token');
+      });
+
+      it('should return 401 error', () => {
+        assert.equal(response.status, 401);
+      });
+    });
+
+    context('when good token was provided', () => {
+      let response, userDoc;
+
+      before(async () => {
+        userDoc = await User.create({
+          username: 'Test',
+          email: 'test@email.com',
+          passwordHash: 'password_hash',
+        });
+
+        const token = jwt.sign({ id: userDoc._id }, process.env.JWT_SECRET, {
+          expiresIn: 2 * 24 * 60 * 60,
+        });
```

Comment:
could you also add this token to the `tokens` array of user document? Also in authorization middleware please change `const user = await userModel.findById(userId);` to `const user = await userModel.findOne({ _id: userId, tokens: token});` Sorry, forgot about it when made review of auth middleware logic

Reply:
Done